data-analysis/12-generate-data.ipynb

```diff
@@ -0,0 +1,1958 @@
+       "3                  5  [24995bc0-bde0-47d4-af36-3000bb9eb7a0, babeab2...   12   \n",
+       "4                  2  [9e5b4d43-2aff-4318-b2e0-2aa8b2a2a277, d76c707...   13   \n",
+       "...              ...                                                ...  ...   \n",
+       "651035             2  [1ee927e7-24bf-4c5c-ba0a-80b0a040a511, 9ea62f1...   17   \n",
+       "651036             2  [650064a3-f08d-4a42-a222-02b936a2c6b3, b55164e...   18   \n",
+       "651037             2  [f172f6f3-87e0-41ae-850d-8f8d92340a03, d32e2ce...   19   \n",
+       "651038             2  [3495e353-a548-4b9e-a037-9db85f0c7577, c11efca...   20   \n",
+       "651039             4  [f27a1bba-87fe-4bb3-ad9f-728e9a11e7ae, 04a2e72...   21   \n",
+       "\n",
+       "           day      date                          name  \\\n",
+       "0       Monday  20210503                   King-Mendez   \n",
+       "1       Monday  20210503                   King-Mendez   \n",
+       "2       Monday  20210503                   King-Mendez   \n",
+       "3       Monday  20210503                   King-Mendez   \n",
+       "4       Monday  20210503                   King-Mendez   \n",
+       "...        ...       ...                           ...   \n",
+       "651035  Friday  20210528  Gordon, Holmes and Rodriguez   \n",
+       "651036  Friday  20210528  Gordon, Holmes and Rodriguez   \n",
+       "651037  Friday  20210528  Gordon, Holmes and Rodriguez   \n",
+       "651038  Friday  20210528  Gordon, Holmes and Rodriguez   \n",
+       "651039  Friday  20210528  Gordon, Holmes and Rodriguez   \n",
+       "\n",
+       "                               address           city region country  \n",
+       "0       322 Elizabeth Shores Suite 970     Nancyville     AB      CA  \n",
+       "1       322 Elizabeth Shores Suite 970     Nancyville     AB      CA  \n",
+       "2       322 Elizabeth Shores Suite 970     Nancyville     AB      CA  \n",
+       "3       322 Elizabeth Shores Suite 970     Nancyville     AB      CA  \n",
+       "4       322 Elizabeth Shores Suite 970     Nancyville     AB      CA  \n",
+       "...                                ...            ...    ...     ...  \n",
+       "651035                0896 Dennis Mill  Port Jeantown     SK      CA  \n",
+       "651036                0896 Dennis Mill  Port Jeantown     SK      CA  \n",
+       "651037                0896 Dennis Mill  Port Jeantown     SK      CA  \n",
+       "651038                0896 Dennis Mill  Port Jeantown     SK      CA  \n",
+       "651039                0896 Dennis Mill  Port Jeantown     SK      CA  \n",
+       "\n",
+       "[651040 rows x 10 columns]"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df = pd.DataFrame(audiences)\n",
+    "df.day = pd.Categorical(df.day, categories=['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'], ordered=True)\n",
```

Comment:
Maybe explaining what `pd.Categorical()` function does, or alternatives for how it can be purposed. (Unless seen in a previous training session) 🤷‍♀️

Reply:
yes, good point. should elaborate on that in written material.

perhaps in the live session, we can actually carry through this _without_ the ordered categorical column, until it generates the first heatmap which shows weirdly ordered week days, then remedy it with the category column